crypto/x509/x509_vfy.c

```diff
@@ -359,8 +359,6 @@ static int check_issued(ossl_unused X509_STORE_CTX *ctx, X509 *x, X509 *issuer)
      * SUBJECT_ISSUER_MISMATCH just means 'x' is clearly not issued by 'issuer'.
      * Every other error code likely indicates a real error.
      */
-    if (err != X509_V_ERR_SUBJECT_ISSUER_MISMATCH)
-        ctx->error = err;
```

Comment:
Perhaps drop the comment too?

Reply:
If the aim was just to revert #14094, I would have dropped this comment, too, and further simplified this function again.
Yet the comment will be helpful again with #18761.